### PR TITLE
[W5][W15-2] Johnson Ho Chong Xiang

### PR DESCRIPTION
**Pull request by jhocx**

### CODE REVIEW

**Comment by jamessspanggg** [comment]
Good job on adding the count command and its test cases! However, the commit messages can be in present tense.

File: src/seedu/addressbook/parser/Parser.java [hunk 1]
```diff
@@ -11,17 +11,7 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import seedu.addressbook.commands.AddCommand;
```
Comment (gohyinhao) [inline src/seedu/addressbook/parser/Parser.java:14]: Good job on implementing the count feature! But according to the coding standards, imported classes should always be stated explicitly!

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -72,6 +72,11 @@ Examples:
 Shows a list of all persons, along with their non-private details, in the address book. +
 Format: `list`
 
+== Counting number of persons in the address book : `count`
+
+Returns the number of persons currently stored in the address book. +
```
Comment (leeyh20) [inline docs/UserGuide.adoc:77]: Nice documentation!

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (leeyh20) [inline src/seedu/addressbook/parser/Parser.java:14]: Agreed that imported classes should always be stated explicitly. Also do read Intellij's documentation about using auto import.

File: test/java/seedu/addressbook/commands/CountCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.person.*;
+
+import java.util.HashSet;
+
+public class CountCommandTest {
+
+    AddressBook testAddressBook = new AddressBook();
+
+    @Test
```
Comment (leeyh20) [inline test/java/seedu/addressbook/commands/CountCommandTest.java:14]: Good job adding JUnit Tests!

File: test/java/seedu/addressbook/commands/CountCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                    new Email("C@gmail.com", true),
+                    new Address("C", true),
+                    new HashSet<>())
+            );
+
+            CountCommand testCommand = new CountCommand();
+            testCommand.setData(testAddressBook,null);
+            CommandResult testResult = testCommand.execute();
+
+            Assert.assertEquals(CountCommand.MESSAGE_SUCCESS + 3, testResult.feedbackToUser);
```
Comment (leeyh20) [inline test/java/seedu/addressbook/commands/CountCommandTest.java:45]: This is a good example for testing a success case for 3 people. How about adding tests for the case where the address book is empty?

File: src/seedu/addressbook/parser/Parser.java (hunk 3 above)
Comment (leeyh20) [reply src/seedu/addressbook/parser/Parser.java:14]: If your IDE is automatically converting this, please do change the settings to prevent it from doing so.

File: test/java/seedu/addressbook/commands/CountCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import seedu.addressbook.data.person.*;
+
+import java.util.HashSet;
+
+public class CountCommandTest {
+
+    AddressBook testAddressBook = new AddressBook();
+
+    @Test
+    public void testCountCommandOutput() {
```
Comment (leeyh20) [inline test/java/seedu/addressbook/commands/CountCommandTest.java:15]: Do look at the name of the tests for other commands. This is not really following the naming convention of the tests.

**Review by leeyh20 — APPROVED** [review]
Overall, good work for adding a small but useful enhancement!

You could also provide more information about your PR in the PR description (what feature you have created, describe and justify the way you implemented the feature)

**Comment by leeyh20** [comment]
@jhocx Remember to close PR after reading, thanks!
